### PR TITLE
Add V2 structures & finalize V3 spec

### PR DESCRIPTION
**Pull request by TroyDota**

### CODE REVIEW

File: structures/v2/type.cosmetic.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	// Shape of a radial gradient, when the paint is of RADIAL_GRADIENT type
+	Shape string `json:"shape,omitempty" bson:"shape,omitempty"`
+	// URL of an image, when the paint is of BACKGROUND_IMAGE type
+	ImageURL string `json:"image_url,omitempty" bson:"image_url,omitempty"`
+	// Deprecated: Properties for a drop shadow
+	DropShadow CosmeticPaintDropShadow `json:"drop_shadow,omitempty" bson:"drop_shadow,omitempty"`
+	// A list of drop shadows. There may be any amount, which can be stacked onto each other
+	DropShadows []CosmeticPaintDropShadow `json:"drop_shadows,omitempty" bson:"drop_shadows,omitempty"`
+	// Animation instructions, if any
+	Animation CosmeticPaintAnimation `json:"animation,omitempty" bson:"animation,omitempty"`
```
Comment (AnatoleAM) [inline structures/v2/type.cosmetic.go:74]: You can remove this it's unused

File: structures/v3/structures.go [hunk 2]
```diff
@@ -64,32 +59,3 @@ var (
 )
 
 type ObjectID = primitive.ObjectID
-
-var (
-	DefaultRoles     []*Role = nil
-	defaultRolesLock sync.Mutex
-)
-
-func FetchDefaultRoles(ctx context.Context, mngo mongo.Instance) []*Role {
```
Comment (AnatoleAM) [inline structures/v3/structures.go:73]: This is removed, but do we have a better idea for implementing default permissions?

Comment (TroyDota) [reply structures/v3/structures.go:73]: Uh, we should likely cache it in redis.

Comment (AnatoleAM) [reply structures/v3/structures.go:73]: Right yeah, that's pretty true
yeah that's true

**Pull request merged**